File: algorithm/unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        alphabet_morse = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--",
                          "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]
        alphabets = 'abcdefghijklmnopqrstuvwxyz'
        alphabet_morse_dict = {alphabet: morse for alphabet, morse in zip(alphabets, alphabet_morse)}
        res = set()
        for word in words:
            res.add(''.join(alphabet_morse_dict[w] for w in word))
        return len(res)


if __name__ == '__main__':
    print(Solution().uniqueMorseRepresentations(["gin", "zen", "gig", "msg"]))

File: Ruby/unique-morse-code-words.rb
# @param {String[]} words
# @return {Integer}
def unique_morse_representations(words)
  word_map = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]
  morse = []
  words.each do |item|
    morse_res = ""
    item.each_char do |item_char|
      morse_res << word_map[item_char.ord - 97]
    end
    if !morse.include?(morse_res)
      morse.push(morse_res)
    end
  end
  return morse.size
end

words = ["gin", "zen", "gig", "msg"]
times = unique_morse_representations(words)
p times